meta.go

```diff
@@ -1,39 +1,224 @@
+	} else {
+		metaJson, _ := readFile(metaFilePath)
+		// Exist meta.json
+		if len(metaJson) != 0 {
+			err = json.Unmarshal(metaJson, &previousMeta)
+			if err != nil {
+				return err
+			}
+		} else {
+			// Exist meta.json but it is empty
+			previousMeta = make(map[string]interface{})
+		}
+	}
+
+	key, parsedValue := parseMetaValue(key, value, previousMeta)
+	previousMeta[key] = parsedValue
+
+	resultJson, err := json.Marshal(previousMeta)
+
+	err = writeFile(metaFilePath, resultJson, 0666)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-var cleanExit = func() {
+// Parse arguments of meta-cli to JSON
+func parseMetaValue(key string, value string, previousMeta interface{}) (string, interface{}) {
+	for position, char := range key {
+		if string([]rune{char}) == "[" {
+			nextChar := key[position+1]
+			if nextChar == []byte("]")[0] {
+				// Value is array
+				var metaValue [1]interface{}
+				key = key[0:position] + key[position+2:] // Remove bracket[] from key
+				key, metaValue[0] = parseMetaValue(key, value, previousMeta)
+				return key, metaValue
+			} else {
+				// Value is array with index
+				var i int
+				for i = position + 1; ; i++ {
+					_, err := strconv.Atoi(string(key[i])) // Check the next char is integer
+					if err != nil {
+						break
+					}
+				}
+				metaIndex, _ := strconv.Atoi(key[position+1 : i]) // e.g. if array[10], get "10"
+				key = key[0:position] + key[i+1:]                 // Remove bracket[num] from key
+
+				// Convert previousMeta Interface to Map
+				previousMetaValue := reflect.ValueOf(previousMeta)
+				var previousMetaMap map[string]interface{}
+				previousMetaMap = make(map[string]interface{})
+				var previousKey string
+				if previousMetaValue.Kind() == reflect.Map {
+					for _, k := range previousMetaValue.MapKeys() {
+						previousKey, _ = k.Interface().(string)
+						previousMetaMap[previousKey] = previousMetaValue.MapIndex(k).Interface()
+					}
+				}
+
+				var metaValue []interface{}
+				// previousMetaMap[previousKey] is empty or string, create array with null except "value"
+				if previousMetaMap[previousKey] == nil || reflect.ValueOf(previousMetaMap[previousKey]).Kind() == reflect.String {
+					metaValue = make([]interface{}, metaIndex+1)
+					key, metaValue[metaIndex] = parseMetaValue(key, value, previousMetaMap[previousKey])
+					return key, metaValue
+				} else {
+					metaValue = make([]interface{}, metaIndex+1)
+					previousObject := reflect.ValueOf(previousMetaMap[previousKey])
+					if metaIndex+1 > previousObject.Len() {
+						metaValue = make([]interface{}, metaIndex+1)
+						key, metaValue[metaIndex] = parseMetaValue(key, value, nil)
+					} else {
+						metaValue = make([]interface{}, previousObject.Len())
+						key, metaValue[metaIndex] = parseMetaValue(key, value, previousObject.Index(metaIndex).Interface())
+					}
+					// Insert previousValues to a[] when previousObject is Array
+					if previousObject.Kind() == reflect.Slice {
```

Comment:
Is there a way to clone the `previousObject` array instead of having to go through the array and copy the old values?